src/main/java/org/skriptlang/skript/lang/errors/RuntimeErrorProducer.java

```diff
@@ -0,0 +1,158 @@
+	 *
+	 * @param message The text to display as the error message.
+	 */
+	default void warning(String message) {
+		String formatted = toFormattedErrorString(Level.WARNING, message, getNode(), toUnderline());
+		Skript.getRuntimeErrorManager().warning(getNode(), formatted);
+	}
+
+	String CONFIG_NODE = "skript command.reload";
+	ArgsMessage WARNING_DETAILS = new ArgsMessage(CONFIG_NODE + ".warning details");
+	ArgsMessage ERROR_DETAILS = new ArgsMessage(CONFIG_NODE + ".error details");
+	ArgsMessage OTHER_DETAILS = new ArgsMessage(CONFIG_NODE + ".other details");
+	ArgsMessage ERROR_INFO = new ArgsMessage(CONFIG_NODE + ".runtime.error");
+	ArgsMessage WARNING_INFO = new ArgsMessage(CONFIG_NODE + ".runtime.warning");
+	ArgsMessage LINE_INFO = new ArgsMessage(CONFIG_NODE + ".runtime.line info");
+
+	/**
+	 * Generates a formatted string based on the provided log level, message, node information, and text to underline.
+	 *
+	 * @param level The severity level of the log message.
+	 * @param message The message to be included in the log.
+	 * @param node The node associated with this log message.
+	 * @param toUnderline The text in the node's line content that should be underlined in the log message.
+	 * @return A formatted string that includes metadata such as the script filename, error message,
+	 * 			and the specific line that caused the error.
+	 */
+	private @NotNull String toFormattedErrorString(@NotNull Level level, String message, Node node, String toUnderline) {
+		// Replace configured messages chat styles without user variables
+		// todo: add lang entries, handle error vs warning
+
+
+		ArgsMessage details;
+		ArgsMessage info;
+		if (level.intValue() == Level.WARNING.intValue()) { // warnings
+			details = WARNING_DETAILS;
+			info = WARNING_INFO;
+		} else if (level.intValue() == Level.SEVERE.intValue()) { // errors
+			details = ERROR_DETAILS;
+			info = ERROR_INFO;
+		} else { // anything else
+			details = OTHER_DETAILS;
+			info = WARNING_INFO;
+		}
+
+		String skriptInfo = replaceNewline(Utils.replaceEnglishChatStyles(info.getValue() == null ? info.key : info.getValue()));
+		String errorInfo = replaceNewline(Utils.replaceEnglishChatStyles(details.getValue() == null ? details.key : details.getValue()));
+		String lineInfo = replaceNewline(Utils.replaceEnglishChatStyles(LINE_INFO.getValue() == null ? LINE_INFO.key : LINE_INFO.getValue()));
+
+		if (node == null)
+			return "effect command produced error"; // TODO
+
+		Config c = node.getConfig();
+		String from = this.getClass().getAnnotation(Name.class).value().trim().replaceAll("\n", "");
+
+		return
+			String.format(skriptInfo, c.getFileName(), from, getSyntaxType()) +
+			String.format(errorInfo, message.replaceAll("§", "&")) +
+			String.format(lineInfo, node.getLine(), node.save().trim().replaceAll("§", "&").replaceAll(toUnderline, "§f§n" + toUnderline + "§7"));
+	}
+
+	@Contract(pure = true)
+	private @NotNull String getSyntaxType() {
```

Comment:
It might be worth adding this as a method onto SyntaxElement. The element classes can have default implementations (maybe final)